sql/2021/privacy/nb_websites_using_each_cmp.sql

```diff
@@ -0,0 +1,47 @@
+#standardSQL
+# Percent of websites using a specific CMP (Based on wappalyzer 'Cookie compliance' category)
+# Alternatively, `core_web_vitals.technologies` could be used, but then we do not have
+#  access to the total number of websites
+
+WITH apps AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url,
+    IF(category = "Cookie compliance", app, "") AS cmp_app
+  FROM `httparchive.technologies.2021_07_01_*`
+  GROUP BY
+    client,
+    url,
+    cmp_app
+),
+
+base AS (
+  SELECT
+    client,
+    url,
+    cmp_app,
+    COUNT(DISTINCT url) OVER (PARTITION BY client) AS total_pages,
+    COUNT(DISTINCT url) / COUNT(DISTINCT url) OVER (PARTITION BY client) AS pct_pages_with_cmp
+  FROM
+    apps
+  GROUP BY
+    client,
+    url,
+    cmp_app
+)
```

Comment:
Total pages when grouped bu the same url? Does this make sense?

Reply:
While it's a window function there is no calculation error, but it may be simplified.